### PR TITLE
Provide end-to-end Triton server build

### PR DESCRIPTION
**Pull request by wphicks**
Provide a Dockerfile, build script, and associated documentation for building a Triton server image with FIL backend from any Ubuntu 20.04-based image

### CODE REVIEW

File: ops/e2e.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+    pkg-config \
+    python3-dev \
+    python3-libnvinfer-dev \
+    python3-opencv \
+    python3-pip \
+    python3-setuptools \
+    software-properties-common \
+    tensorrt \
+ && rm /tmp/tensorrt.deb \
+ && rm -rf /var/lib/apt/lists/*
```
Comment (drobison00) [inline ops/e2e.Dockerfile:96]: I think you'll want to run `apt clean` at the end of each command chain, that will delete all cached packages as well.

File: ops/e2e.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+    libssl-dev \
+    libboost-dev \
+    libcurl4-openssl-dev \
+    patchelf \
+    rapidjson-dev \
+    unzip \
+    uuid-dev \
+    wget \
+    zlib1g-dev \
+ && rm -rf /var/lib/apt/lists/*
```
Comment (drobison00) [inline ops/e2e.Dockerfile:126]: apt clean

File: ops/e2e.Dockerfile [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+# Install CMake
+RUN wget -O - https://apt.kitware.com/keys/kitware-archive-latest.asc 2>/dev/null \
+    | gpg --dearmor - \
+    | tee /etc/apt/trusted.gpg.d/kitware.gpg >/dev/null \
+ && apt-add-repository 'deb https://apt.kitware.com/ubuntu/ focal main' \
+ && apt-get update \
+ && apt-get install -y --no-install-recommends \
+    cmake-data=3.18.4-0kitware1ubuntu20.04.1 \
+    cmake=3.18.4-0kitware1ubuntu20.04.1 \
+ && rm -rf /var/lib/apt/lists/*
```
Comment (drobison00) [inline ops/e2e.Dockerfile:137]: apt clean

File: ops/e2e.Dockerfile [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+    cuda-cudart-11-2=11.2.146-1 \
+    cuda-compat-11-2 \
+ && apt-get install -y \
+    cuda-libraries-11-2 \
+ && ln -s cuda-11.2 /usr/local/cuda \
+ && apt-get install -y --no-install-recommends \
+    libcudnn8=$CUDNN_VERSION-1+cuda11.3 \
+    libcudnn8-dev=$CUDNN_VERSION-1+cuda11.3 \
+ && apt-mark hold libcudnn8 \
+ && rm -rf /var/lib/apt/lists/*
```
Comment (drobison00) [inline ops/e2e.Dockerfile:62]: apt clean

File: ops/e2e.Dockerfile [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+    python3-pip \
+    python3-setuptools \
+    software-properties-common \
+    tensorrt \
+ && rm /tmp/tensorrt.deb \
+ && rm -rf /var/lib/apt/lists/*
+
+RUN pip3 install --upgrade pip && \
+    pip3 install --upgrade wheel setuptools docker && \
+    pip3 install grpcio-tools grpcio-channelz
```
Comment (drobison00) [inline ops/e2e.Dockerfile:100]: Will want to add `pip cache purge` here

**Review by drobison00 — CHANGES_REQUESTED** [review]
Overall looks good, I think it just needs a couple `apt clean` and `pip cache purge` commands added to the end of some command chains to keep the size down.

**Comment by wphicks** [comment]
Thanks so much! Updated to add apt and pip cache purging accordingly

**Review by aroraakshit — APPROVED** [review]
Looks good!

**Review by GuanLuo — APPROVED** [review]

**Pull request merged**